internal/controller/cloudsyncmgr/cloudsync.go

```diff
@@ -65,37 +65,50 @@ func (c *CloudSyncImpl) InitiateCloudSync(isCloudSet string) (err error) {
 		if strings.Compare(strings.ToLower(isCloudSet), "true") == 0 {
 			log.Println("CloudSync init set")
 			isCloudSyncSet = true
+			//Intialize the client and hashmap storing client data
+			mqttmgr.SetClient(nil)
+			mqttmgr.InitClientData()
 		}
 	}
 	return nil
 }
 
-//StartCloudSync is used to start the sync by connecting to the broker
-func (c *CloudSyncImpl) StartCloudSync(host string) (err error) {
-	if isCloudSyncSet && len(host) > 0 {
-		log.Info("Starting the CLoudsync Mgr")
-		go mqttmgr.StartMQTTClient(host)
-	}
-	return
-}
-
 // RequestCloudSyncConf is  configuration request handler
-func (c *CloudSyncImpl) RequestCloudSyncConf(message mqttmgr.Message, topic string, clientID string) string {
+func (c *CloudSyncImpl) RequestCloudSyncConf(host string, clientID string, message mqttmgr.Message, topic string) string {
 	log.Info(logPrefix, "Publishing the data to the cloud")
-	mqttClient = mqttmgr.GetClient()
-	mqttmgr.SetClientID(clientID)
 	resp := ""
-	if mqttClient.IsConnected() {
+	var wg sync.WaitGroup
+	if isCloudSyncSet {
+		log.Info("Starting the Cloudsync Mgr")
+		if len(host) == 0 {
+			return "No broker host defined"
+		}
+		wg.Add(1)
+		errs := make(chan string, 1)
+		go func() {
+			errs <- mqttmgr.StartMQTTClient(host, clientID)
+			resp = <-errs
+			log.Info("The error message received is ", resp)
```

Comment:
Please display this log if it's not empty.

Reply:
This wwas testing log and hence will be removed

Reply:
Done!